UpdateExpiredCgn/handler.ts

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * This function is not intended to be invoked directly. Instead it will be
+ * triggered by an orchestrator function.
+ *
+ * Before running this sample, please:
+ * - create a Durable orchestration function
+ * - create a Durable HTTP starter function
+ * - run 'yarn add durable-functions' from the wwwroot folder of your
+ *   function app in Kudu
+ */
+
+import { Context } from "@azure/functions";
+import { TableService } from "azure-storage";
+import * as date_fns from "date-fns";
+import * as df from "durable-functions";
+import { isLeft, toError } from "fp-ts/lib/Either";
+import { tryCatch } from "fp-ts/lib/TaskEither";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { StatusEnum as CgnActivatedStatusEnum } from "../generated/definitions/CgnActivatedStatus";
+import { StatusEnum as CgnExpiredStatusEnum } from "../generated/definitions/CgnExpiredStatus";
+import { StatusEnum as CgnRevokedStatusEnum } from "../generated/definitions/CgnRevokedStatus";
+import { OrchestratorInput } from "../UpdateCgnOrchestrator";
+import {
+  makeUpdateCgnOrchestratorId,
+  terminateOrchestratorTask
+} from "../utils/orchestrators";
+import { getExpiredCgnUsers } from "./table";
+
+const finish = () => Promise.resolve(void 0);
+const ORCHESTRATION_TERMINATION_REASON = "An highest priority CGN update orchestrator needs to start" as NonEmptyString;
+
+export const getUpdateExpiredCgnHandler = (
+  tableService: TableService,
+  cgnExpirationTableName: NonEmptyString,
+  logPrefix: string = "UpdateExpiredCgnHandler"
+) => async (context: Context): Promise<void> => {
+  const today = date_fns.format(Date.now(), "yyyy-MM-dd");
+
+  const errorOrExpiredCgnUsers = await getExpiredCgnUsers(
+    tableService,
+    cgnExpirationTableName,
+    today
+  ).run();
+
+  if (isLeft(errorOrExpiredCgnUsers)) {
+    context.log.verbose(
+      `${logPrefix}|ERROR=${errorOrExpiredCgnUsers.value.message}`
+    );
+    return finish();
+  }
+
+  const expiredCgnUsers = errorOrExpiredCgnUsers.value;
+  context.log.info(
+    `${logPrefix}|Processing ${expiredCgnUsers.length} expired CGNs`
+  );
+
+  const client = df.getClient(context);
+  // trigger an update orchestrator for each user's CGN that expires
+  expiredCgnUsers.forEach(async fiscalCode => {
```

Comment:
this will execute task in parallel, you'd better sequence taskEithers here to compute them serially

Reply:
I'm wondering about sequencing a lot of taskEithers since it would raise an error if it reaches max limit. I'd rather avoid to manage chunks as in this [PR](https://github.com/pagopa/io-functions-admin/blob/5b65a4666364dc990f487a885a7519710ba525e3/UpdateMunicipalities/handler.ts#L49) ( a long time ago ) :)

Reply:
that's right but this may rise a lot of parallel executions with a lot of outgoing connection (to the db or whatsoever)

Reply:
Nevermind, I got your point @gunzip :)